### PR TITLE
Fixed #23387 -- Keep "Save as new" button after validation errors

### PR DESCRIPTION
**Pull request by adamzap**
This PR addresses @timgraham's comments on #4265.

### CODE REVIEW

**Comment by timgraham** [comment]
buildbot, add to whitelist.

File: tests/admin_views/tests.py [hunk 1]
```diff
@@ -1151,18 +1151,60 @@ def test_save_as_duplication(self):
...
+    def test_save_as_new_with_validation_errors(self):
+        """
+        Ensure that when you click "Save as new" and have a validation error,
+        you only see the "Save as new" button and not the other save buttons,
+        and that only the "Save as" button is visible.
+        """
+        response = self.client.post(reverse('admin:admin_views_person_change', args=(self.per1.pk,)), {
+            '_saveasnew': '',
+            'gender': 'invalid',
+            '_addanother': 'fail'
```
Comment (timgraham) [inline tests/admin_views/tests.py]: please include the trailing comma on the last item in a dictionary so if more items are added we don't have to modify this line again

File: tests/admin_views/tests.py [hunk 2]
```diff
@@ -1151,18 +1151,60 @@ def test_save_as_duplication(self):
...
+        })
+
```
Comment (timgraham) [inline tests/admin_views/tests.py]: chop newline for consistency with other tests

**Comment by adamzap** [comment]
There you go @timgraham 

Thanks for the feedback

**Comment by timgraham** [comment]
merged in 0894643e40327e48397573b7844585618200442b, thanks!

**Pull request closed without merging**